lib/node_types/esp/src/device.h

```diff
@@ -47,10 +136,38 @@ class Callback {
 };
 
 
-// This is a chained list of callbacks working on a device
-// If called it traverses the chain and all it's subchain elements and
-// returns only true if all subcalls returned true - else it will return false
-// the call back operates on a device as parameter
+/**
+ * @class Callback_Link
+ * @brief Linked list of callbacks for processing pipelines
+ * 
+ * PURPOSE
+ * =======
+ * Allows multiple callbacks to be chained together in a processing pipeline.
+ * All callbacks in the chain must return true for the chain to succeed.
+ * 
+ * EXAMPLE USE CASES
+ * =================
+ * 1. Filter Chain:
+ *    - First callback: Validate sensor reading is in valid range
+ *    - Second callback: Apply calibration correction
+ *    - Third callback: Average with previous readings
+ * 
+ * 2. Trigger Chain:
+ *    - First callback: Check if temperature above threshold
+ *    - Second callback: Send alert if not already sent
```

Comment:
This one is slightly confusing as alerts will be sent automatically if filter chain returns true and the changed value. Replace second and third with turn on signalling led and turn on cooling system relay